### ISSUE TITLE
Breaking Contributing.md into sub-sections in order to define processes more clearly

### ISSUE THREAD

**Issue by pavlenex** [btcpayserver/btcpayserver-doc#671]
Ubuntu has a very nice documentation of processes for new contributors. I think the way we now have them is a bit too generic, perhaps we can break it down into specific sections and provide clear processes on how to contribute to a certain part of btcpay dev / doc / design/ seo / etc

I think for new contributors to get involved they need clear instructions and well-documented process, I think we can break that down in docs

http://ubuntu-manual.org/getinvolved

I would probably remove  Hosting Providers or keep such as on a generic page.
Contributing.md should probably be a page to help encourage new contributors where as all the sub-pages should define simply the process and tools for each "team".

If you're aware of similar pages in foss, let's share them to brainstorm a solution that would fit our community.

**Comment by Zaxounette** [comment]
I'm willing to draft something up tomorrow in the morning,

**Comment by britttttk** [comment]
I think WriteStack should be changed, to WriteSoftware... it does not make sense to me how it is now. I also think that Write in the document title is unnecessary and is confusing, WriteDirectory and WriteStack are examples of this...

docs/Contribute/ContributeWrite/WriteDocs.md  words are repeating here, seems redundant

**Comment by pavlenex** [comment]
I tend to agree with @britttttk , I mentioned to Zax that writers are not likely to do a software change, even if it's a typo.

Which title are you referring to, the filenames?

**Comment by britttttk** [comment]
Actually I'm thinking about it now, I understand the reason for the Write in the titles.. this makes sense to me. We would also then always know that it's a contribution doc by the Write in the title.

**Comment by Zaxounette** [comment]
> I think WriteStack should be changed, to WriteSoftware... it does not make sense to me how it is now.

Sure I can do that if it's confusing.
But keep in mind, it writing **for** the software, not writing **the** software.

**Comment by Zaxounette** [comment]
> > I think WriteStack should be changed, to WriteSoftware... it does not make sense to me how it is now.
> 
> Sure I can do that if it's confusing.
> But keep in mind, it writing **for** the software, not writing **the** software, that'll be covered in ContributeDev.

FYI, done in #684

**Comment by Zaxounette** [comment]
@britttttk , following your comment and link to your old branch ([here](https://github.com/btcpayserver/btcpayserver-doc/pull/684#issuecomment-683177525))
I'd like to get it into the docs for the ContributeCode section (yes, renamed it, not ContributeDev anymore) even if ther's still some TODO's at the bottom.

I'm pretty sure if that doc gets in, i'll have another folder `ContributeCode` with a readme.md for general stuff + your file as a `step by step for beginning to code` doc.

How do you wish to proceed ? You commit it, or do I copy it and add it in a PR on my side ? Wouldn't wan't to steal the doc, which is great !

If you commit it, just create a folder `ContributeCode` inside `Docs/Contribute` and i'll manage from there.

**Comment by britttttk** [comment]
@Zaxounette I just updated the document a bit. I think that we should copy from it, rather than link to it, in case I accidentally delete that branch :D If that's what you mean. 

If you want to keep working on this ContributeCode and grab what you like from that doc that's fine with me, not worried about anyone using it! I can make a PR for the testers gude #676 and move some things around in another PR to make things more clear. 

The last section TODO is just for a reminder for me, some ideas for futher documentation. We can make issues for those instead of keeping them there now this is being worked on. 

Let me know what you think, I'm planning to work on testersDoc later today.

**Comment by Zaxounette** [comment]
>  I think that we should copy from it, rather than link to it, in case I accidentally delete that branch :D

I meant do you wan't to add it yourself via a PR for the PR/commit count, or do I PR and commit that doc myself, thus the "stealing" part :p 



What do you think about this folder setup :

Folder name: `ContributeDev` (Yes, maybe let it be dev after all)

Inside `ContributeDev`:
 `ContributeDev/readme.md` - Generalities for coding and testing + lining to following docs.
`ContributeDev/ContributeDevCode` - Direct coding related. How to start coding and whatnot. Based on the actual contribute.md#Developpement + your `old branch doc` + additions
`ContributeDev/ContributeDevTest` - Not direct code writing related, but code testing related. That could be your upcomming `testersDoc` ?

**Comment by Zaxounette** [comment]
Checkbox summary, for my peace of mind, since there's starting to be a lot of stuff and don't want to lose track..

- [X] Create `Contribute` folder
- [ ] Create & write `Contribute/README.md`
- [X] Create & write `Contribute/ContributeTranslate.md` + add to config.js
- [ ] Create & write `Contribute/ContributeDesign.md` (WIP, @Zaxounette )
- [ ] Create & write `Contribute/ContributeMisc.md` (Miscellaneous, WIP, @Zaxounette )
- [ ] Edit config.js to include remaining files.

----
- [X] Create `ContributeWrite` folder in `Contribute` 
- [X] Create & write `ContributeWrite/README.md`
- [X] Create & write `ContributeWrite/WriteBlog.md`
- [X] Create & write `ContributeWrite/WriteDocs.md`
- [X] Create & write `ContributeWrite/WriteSoftware.md`
- [X] Edit config.js to include above files.
----
- [X] Create `ContributeDev` folder in `Contribute`
- [ ] Create & write `ContributeDev/README.md`
- [ ] Create & write `ContributeDev/ContributeDevCode.md` (WIP, @britttttk )
- [ ] Create & write `ContributeDev/ContributeDevTest.md` (WIP, @britttttk )
- [ ] Edit config.js to include above files.

----

FYI, inside `ContributeMisc.md` I intend to include SEO, Marketing, Video & Graphics (could be considered partly the same as marketing ?), Spreading the Word, Feedback since I feel those subjects for the time being wouldn't require a full page.

Can be expanded if needed, for example Marketing, once a team is structured around that subject and info can be added.

**Comment by Zaxounette** [comment]
@britttttk 
#696 being merged, do I check the `Create & write ContributeDev/ContributeDevTest.md` box ?
I'm unable to review since not technical enough. Are you satisfied with it and do you consider it done ?

**Comment by Zaxounette** [comment]
Post-it note: Add link to https://github.com/btcpayserver/btcpayserver/pull/1627 doc once that PR is merged.

Should probably add it in `ContributeDev/README.md`.

**Comment by britttttk** [comment]
@Zaxounette 
>check the Create & write ContributeDev/ContributeDevTest.md box ? Same for ContributeDevTest.md ?

Let's mark these as complete (new docs have been created) and now we can just do the normal process of refining and making updates and formatting as we go. 

>I might just add a table of content to the ContributeDevCode and DevTest docs, since there's a lot to read.

Any formatting/updates are appreciated!

**Comment by Zaxounette** [comment]
Another addition to make:

https://github.com/btcpayserver/btcpayserver-doc/pull/704#issuecomment-691387429

It is important that links work in local. I'll add a sentence on that and a link in contributewrite.

**Comment by pavlenex** [comment]
LGTM, will take some time to re-read as a whole and then do a separate issue/PR if anything is missing

Awesome work on this both Zax and Brit 🚀
